app/assets/javascripts/host_edit.js

```diff
@@ -221,23 +221,45 @@ function hostgroup_changed(element) {
   var host_id = $("form").data('id');
   var host_changed = $("form").data('type-changed');
   if (host_id) {
-    if (host_changed ){
-      update_form(element,{data:"&host[id]="+host_id});
-    } else if (host_changed == undefined) { // hostgroup changes parent
-      update_form(element);
-    } else { // edit host
-      set_inherited_value(element);
-      update_puppetclasses(element);
-      reload_host_params();
-    }
+    handleHostgroupChangeEdit(element, host_id, host_changed)
   } else { // a new host
-    reset_explicit_values(element);
+    disableComputeResourceDropdown(true);
+    handleHostgroupChangedNew(element).then(function () {
+      disableComputeResourceDropdown(false);
+    });
+  }
+}
+
+function promise() {
+  var deferred = $.Deferred();
+  deferred.reject();
+  return deferred.promise();
+}
+
+function handleHostgroupChangeEdit(element, host_id, host_changed) {
+  if (host_changed){
+    update_form(element,{data:"&host[id]="+host_id})
+  } else if (host_changed == undefined) { // hostgroup changes parent
+    update_form(element)
+  } else { // edit host
     set_inherited_value(element);
-    // call for form update only if there is a hostgroup selected
-    if ($('#host_hostgroup_id').val() != "") {
-      update_form(element);
-    }
+    update_puppetclasses(element);
+    reload_host_params();
+  }
+}
+
+function handleHostgroupChangedNew(element) {
+  reset_explicit_values(element);
+  set_inherited_value(element);
+  // call for form update only if there is a hostgroup selected
+  if ($('#host_hostgroup_id').val() != "") {
+    return update_form(element);
   }
+  return promise();
+}
+
+function disableComputeResourceDropdown(disable) {
+  $("#host_compute_resource_id").attr("disabled", disable);
```

Comment:
nitpick: `disabled` should use `.prop`, not `.attr`.
And I'm not sure we need a separate function for this one line.